services/dmarc-report/src/calculate-percentages.js

```diff
@@ -4,17 +4,73 @@ const calculatePercentages = ({ fail, pass, passDkimOnly, passSpfOnly }) => {
     0,
   )
 
+  // calculate base percentages
+  const failPercentage = fail <= 0 ? 0 : Number((fail / total) * 100)
+
+  const passPercentage = pass <= 0 ? 0 : Number((pass / total) * 100)
+
+  const passDkimOnlyPercentage =
+    passDkimOnly <= 0 ? 0 : Number((passDkimOnly / total) * 100)
+
+  const passSpfOnlyPercentage =
+    passSpfOnly <= 0 ? 0 : Number((passSpfOnly / total) * 100)
+
+  const percentages = [
+    failPercentage,
+    passPercentage,
+    passDkimOnlyPercentage,
+    passSpfOnlyPercentage,
+  ]
+
+  // do a basic rounding of all percentages
+  const rounded = percentages.map((x) => Math.floor(x))
+
+  // get the sum of the current rounded percentages
+  const afterRoundSum = rounded.reduce((pre, curr) => pre + curr, 0)
+
+  // get the amount of numbers larger or equal to 1
+  const countMutableItems = rounded.filter((x) => x >= 1).length
```

Comment:
This removes numbers that should probably round to 1
using 0.8, 0.8, 0.8, 97.6
it rounds to 0, 0, 0, 100